### ISSUE TITLE
Trouble with OS X

### ISSUE THREAD

**Issue by Schwalbi** [evilhero/mylar#968]
Hi,
I've dying to get my hands on this software and have been using this guide to help me: http://www.htpcguides.com/install-mylar-mac-osx-usenet-comics/

The Terminal commands were as followed:
Last login: Tue Mar 24 19:55:53 on ttys000
Schwalbis-MacBook-Pro:~ Schwalbi$ xcode-select --install
xcode-select: error: command line tools are already installed, use "Software Update" to install updates
Schwalbis-MacBook-Pro:~ Schwalbi$ sudo easy_install CherryPy
Password:
Sorry, try again.
Password:
Searching for CherryPy
Reading https://pypi.python.org/simple/CherryPy/
Reading http://download.cherrypy.org/cherrypy/3.0.1/
Reading http://download.cherrypy.org/cherrypy/3.0.2/
Reading http://download.cherrypy.org/cherrypy/3.0.3/
Reading http://download.cherrypy.org/cherrypy/3.0RC1/
Reading http://download.cherrypy.org/cherrypy/3.1.0beta3/
Reading http://download.cherrypy.org/cherrypy/3.1.0rc1/
Reading http://download.cherrypy.org/cherrypy/3.1.1/
Reading http://download.cherrypy.org/cherrypy/3.1beta/
Reading http://download.cherrypy.org/cherrypy/3.2.3/
Reading http://sourceforge.net/project/showfiles.php?group_id=56099
Reading http://trac.cherrypy.org/cgi-bin/trac.cgi/wiki/CherryPyDownload
Reading http://www.cherrypy.org
Best match: CherryPy 3.6.0
Downloading https://pypi.python.org/packages/source/C/CherryPy/CherryPy-3.6.0.zip#md5=cd4d1c0f81db02a587da7ee23ac2af16
Processing CherryPy-3.6.0.zip
Writing /tmp/easy_install-pP8fTa/CherryPy-3.6.0/setup.cfg
Running CherryPy-3.6.0/setup.py -q bdist_egg --dist-dir /tmp/easy_install-pP8fTa/CherryPy-3.6.0/egg-dist-tmp-yaMCHG
zip_safe flag not set; analyzing archive contents...
cherrypy._cpmodpy: module references __file__
cherrypy._cptree: module references __file__
cherrypy.lib.covercp: module references __file__
cherrypy.lib.cpstats: module references __file__
cherrypy.lib.gctools: module MAY be using inspect.getframeinfo
cherrypy.lib.profiler: module references __file__
cherrypy.process.plugins: module references __file__
cherrypy.scaffold.__init__: module references __file__
cherrypy.test.benchmark: module references __file__
cherrypy.test.checkerdemo: module references __file__
cherrypy.test.helper: module references __file__
cherrypy.test.modfastcgi: module references __file__
cherrypy.test.modfcgid: module references __file__
cherrypy.test.modpy: module references __file__
cherrypy.test.modwsgi: module references __file__
cherrypy.test.test_caching: module references __file__
cherrypy.test.test_config: module references __file__
cherrypy.test.test_config_server: module references __file__
cherrypy.test.test_core: module references __file__
cherrypy.test.test_logging: module references __file__
cherrypy.test.test_misc_tools: module references __file__
cherrypy.test.test_request_obj: module references __file__
cherrypy.test.test_routes: module references __file__
cherrypy.test.test_session: module references __file__
cherrypy.test.test_states: module references __file__
cherrypy.test.test_static: module references __file__
cherrypy.test.test_virtualhost: module references __file__
cherrypy.test.test_wsgiapps: module references __file__
cherrypy.tutorial.tut01_helloworld: module references __file__
cherrypy.tutorial.tut02_expose_methods: module references __file__
cherrypy.tutorial.tut03_get_and_post: module references __file__
cherrypy.tutorial.tut04_complex_site: module references __file__
cherrypy.tutorial.tut05_derived_objects: module references __file__
cherrypy.tutorial.tut06_default_method: module references __file__
cherrypy.tutorial.tut07_sessions: module references __file__
cherrypy.tutorial.tut08_generators_and_yield: module references __file__
cherrypy.tutorial.tut09_files: module references __file__
cherrypy.tutorial.tut10_http_errors: module references __file__
Adding CherryPy 3.6.0 to easy-install.pth file
Installing cherryd script to /usr/local/bin

Installed /Library/Python/2.7/site-packages/CherryPy-3.6.0-py2.7.egg
Processing dependencies for CherryPy
Finished processing dependencies for CherryPy
Schwalbis-MacBook-Pro:~ Schwalbi$ git clone https://github.com/evilhero/mylar.git -b development /Applications/Mylar
Cloning into '/Applications/Mylar'...
remote: Counting objects: 6561, done.
remote: Total 6561 (delta 0), reused 0 (delta 0), pack-reused 6561
Receiving objects: 100% (6561/6561), 8.77 MiB | 967.00 KiB/s, done.
Resolving deltas: 100% (3458/3458), done.
Checking connectivity... done.
Schwalbis-MacBook-Pro:~ Schwalbi$ python /Applications/Mylar/Mylar.py
Traceback (most recent call last):
  File "/Applications/Mylar/Mylar.py", line 26, in <module>
    import mylar
  File "/Applications/Mylar/mylar/__init__.py", line 50, in <module>
    OS_LANG, OS_ENCODING = locale.getdefaultlocale()
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/locale.py", line 511, in getdefaultlocale
    return _parse_localename(localename)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/locale.py", line 443, in _parse_localename
    raise ValueError, 'unknown locale: %s' % localename
ValueError: unknown locale: UTF-8
Schwalbis-MacBook-Pro:~ Schwalbi$ 

Basically, the last command "python /Applications/Mylar/Mylar.py" generates the error. I went on to discuss this error with the writer of the guide, and he told me that he thought it may have something to do with me running Yosemite as he had never seen a problem of this sort. He uses Mavericks and had no trouble using Mylar. He later tried to use Mylar on Yosemite and succeeded, apparently without the use of CherryPy (which I am unable to uninstall due to problems installing pip); however, Mylar on Mavericks started to collapse and generating the same issue, leading us to think that it wasn't just a Yosemite problem. 

I'm really eager to use this software! 
Please help!

**Comment by evilhero** [comment]
It's a problem with the way yosemite handles the getdefaultlocale command. It's documented on the net (just google 'getdefaultlocale python yosemite' and see). 

From a blurb :
There are numerous workarounds for this problem (turning off the "Set locale environment variables on startup" option in the terminal settings, or adding "export LC_CTYPE=en_US.UTF8" to .bash_profile, selecting a language other than English in the Language & Text settings), but these require additional configuration from the user's side. 

I might not even need the locale option for mylar, but I need to sift through the code to make sure before I remove it.

**Comment by Schwalbi** [comment]
Great! I hope you get to remove the locale option. Please let me know if you ever get around to fixing it. I really want to use this software.

**Comment by evilhero** [comment]
It might not be in the next commit (there's a lot already in the one coming very soon hopefully)  - you could try one of the options I listed above though in the meantime , the easiest being the export LC_CTYPE option since that's the least obtrusive and wouldn't break anything else.

**Comment by Schwalbi** [comment]
Ok. How exactly do I do that?

**Comment by AtariBaby** [comment]
As an alternative, have you considered buying an el cheap computer, or resurrecting a little-used one, and making it into a Windows or Linux-based media server? It would be a project that would span a couple of weekends or more, but then you'd have a dedicated machine that handles all the downloading, and using an OS that's more popular with the usenet crowd.

My home is all Apple, but I built a home media server on Linux. My first one was a $200 laptop.

**Comment by Schwalbi** [comment]
How about if I use Parallel Desktop? Do you think I could get it working there?

**Comment by AtariBaby** [comment]
That should work, though I've never done more than use regular Windows programs with Parallels. There are two modes to Parallels, one is fully seamless, the other is more separated, like you're looking at a full Windows desktop. I would use the more separated mode at first, to make it easier to get support from Windows users. Then you can use the seamless mode once it's running. Or you could use Boot Camp, if you don't mind having to fully boot into Windows and then boot fully into OS X.

The caveats with Parallels are: it eats up resources, so you will lose some power and speed, and it relies heavily on the graphics card, so you may be hearing that jet engine sound from the graphics card fan blasting at full speed. It's also annoying if you're not using a fully paid for version, because of nagging popups and being locked out of their support community.

**Comment by Schwalbi** [comment]
I think it would be more annoying having to turn on Parallels each time I need the new comics. Probably easier to do it the old fashioned way.

**Comment by blindguy** [comment]
How about getting vitrualbox and run a VM?  You could run a windows or small linux distro Vm and install MYLar on that?  That is what I do, much cleaner.

**Comment by Schwalbi** [comment]
Isn't that the same as having a VM on Parallel Desktop?

**Comment by AtariBaby** [comment]
I think you meant to say Bootcamp? Parallels can run all the time, even if you are using the discreet mode. You just flip back and forth between the two OS desktops.

Virtualbox is another way to go, as mentioned, plus you’d have someone here who has done it! It requires good planning and focus on setting it up properly before anything else. It can be rather unforgiving and require you to completely start from scratch if not set up optimally.

**Comment by blindguy** [comment]
No idea about Parallel Desktop.  With virtualbox I have it setup to start the VM on boot so when my box boots, it starts the VM and I can connect to mylar via HTTP.  It's like running two computers at once.

**Comment by AtariBaby** [comment]
Virtualbox and Parallels are very similar. Virtualbox is open source or freeware, right? And has a lot of community support. But it’s a little more rigid and rough around the edges than Parallels.

**Comment by AtariBaby** [comment]
And sorry if I'm hijacking this thread, but if you have a second monitor, with any of the virtual machines, you can have your OS X desktop on one monitor and your virtual desktop on a second monitor.

**Comment by Schwalbi** [comment]
I know I can flip back to Parallels at any time. I don't have it on Boot Camp. But I do have to have it open for the software to work and as you have said that takes a lot of power from the computer. If VirtualBox takes less resources then I think it would be a better choice. But otherwise, I either have to keep Parallels open or open it once per day to check for the new torrents. Either way seem like a hassle since I would much rather prefer it to work on Yosemite. But well, until then I guess this will have to do. I'll post as soon as I get it working.

**Comment by evilhero** [comment]
From a command shell:
nano ~/.bash_profile

add this to the end of the file:
export LC_CTYPE=en_US.UTF8


Ctrl+o to save changes
confirm filename 
ctrl+x to exit

**Comment by AtariBaby** [comment]
I hear you! The solution may be just around the corner for you, for all I know! As an IT person, Mac is now aggressively changing their OS and tightening security, so my instinct would be to go around it. But hopefully evilhero and others can help you find the direct solution. Good luck!

**Comment by evilhero** [comment]
As an aside, I haven't a clue about osx or Mac - the commands I gave will work on either nix or Mac (confirmed it via good ol' google). 

What the OP does with the options provided herein is obviously up to him, but there was a lot of good suggestions provided by the community (that I have no clue about lol)!

**Comment by Schwalbi** [comment]
Hi! I did the LC_CTYPE=en_US.UTF8 fix and now I am currently using Mylar. 
I have been able to get my API key from Comic Vine and I am setting up my watch list for my favourite comics. I'm so excited to see it work.

However, I get the feeling I'm missing something. I'm trying to set it up to download using torrents and have used this settings:
![screen shot 2015-03-26 at 5 38 23 pm](https://cloud.githubusercontent.com/assets/11640052/6858692/02bc908a-d3df-11e4-98a3-ff5eca5e4748.png)
![screen shot 2015-03-26 at 5 38 38 pm](https://cloud.githubusercontent.com/assets/11640052/6858697/0731202c-d3df-11e4-9464-39dafcf4924e.png)

Is this enough? Or do I have to set up something else?

**Comment by evilhero** [comment]
That's enough to get it working - the only thing I would suggest is to up the minimum number of seeders to 1, or else you'll end up with a lot of dead torrents (especially with kat). 

For post-processing to work, it would depend on your torrent client and how you have it configured to handle completed downloads. 

If you move them into separate directory and stop sending, then you can use the folder monitor option within mylar to monitor that 'completed downloads' folder every X minutes and perform post-processing on any issues it finds within that the series is on your watchlist. 

Otherwise, if you move completed torrents manually then you can just do a manual post-processing on a specific folder (Manage tab/manual post-processing). Mylar will only look at cbr/cbz files when doing post-processing. 

Other than those minor things, feel free to ask questions and hopefully you'll be reading some comics in no time ;)

**Comment by Schwalbi** [comment]
I don't mind the whole post-processing. I just want the comics downloaded. I'm not picky with how they are stored or if they have special naming. What I don't understand is how do they start downloading. 

As I've mentioned, I've already added some of my comic book series on the wanted list. Let's use the new Darth Vader comics from Marvel. Issue 003 came out this week; I have downloaded the previous issues the weeks when they were released and don't have them stored on a particular folder. Since issue 003 is on this week's watch list, I was assuming that as soon as I added it as a series, it was going to download a torrent file. However, none of that happened. Did I already missed my chance? Do I have to wait until issue 004? Is there a way to manually start the download?

Sorry for all this questions but I don't seem to find a Mylar for dummies section.

**Comment by Schwalbi** [comment]
I found this in History.
![screen shot 2015-03-26 at 6 05 36 pm](https://cloud.githubusercontent.com/assets/11640052/6859047/cc185d62-d3e2-11e4-8fb3-d2ce1ef5c1a7.png)
It says snatched however it hasn't downloaded any torrent yet.

**Comment by evilhero** [comment]
Ok so it's snatched it, but then it can't send to your torrent watch folder. 

Does your torrent client have a watch folder and is the path you put in for mylar correct? 

Does the user that's running mylar have full access to the given watch directory? 

Did you have the minimum number of seeders set to 1 when it was snatched?

**Comment by Schwalbi** [comment]
The torrent client is currently watching my Downloads folder, as soon as a I put a torrent file there it starts downloading it automatically. I have set the same path in Mylar. I'm the only user in this laptop and have full access. I did not have the minimum number of seeders set to 1, but I clicked retry on both of them and got this reply: "Successful re-download of Darth Vader 3". There are no torrent files anywhere on my computer.

**Comment by evilhero** [comment]
Try restarting mylar and try again - if you make any configuration changes it's usually best to restart mylar just to make sure it has the correct values for items.

Othet than that, in the logs/ folder within the mylar directory there are a bunch of log files - if you check the mylar.log file it might give you more information as to what's happening as the log holds all the debugging information.

**Comment by Schwalbi** [comment]
FUCK YEAH! It's working! It's beautiful! 
It was creating a new folder somewhere else and all torrents were not being downloaded in the watch folder. As soon as I changed that everything worked out fine. As soon as I set a comic as Wanted it downloads the torrent and it automatically adds to the Torrent client. The manual download is also working. Great piece of software you have! 

I just have some last questions. Is Mylar constantly running on the background? Or do I have to check it every now and then for it to start downloading? What about when the laptop restarts?

Thanks for all your help!

**Comment by evilhero** [comment]
haha - glad you got it working :)

It depends on how you started Mylar - if you are running it directly via the python window ('python Mylar.py') then whenever you close that python window, Mylar will stop (and probably not gracefully if you don't stop Mylar first). If you're running Mylar as a daemon/service (which is the recommended way), then it will run in the background until you either shut it down, kill the daemon/service, or shutdown/restart the computer itself.

Mylar will automatically pull down the weekly pull list every Monday (usually it's available sometime between Monday morning and early afternoon), it will then mark any issues it needs on the pull-list as Wanted and put them into the Upcoming section (the bottom part of the Wanted tab). From here, they don't get marked as Wanted on the comic details screen, because the site where Mylar gets all of it's comic data form (ComicVine) doesn't update their new weekly issues until Tuesday, sometimes Wednesday so there's no issue information for Mylar to grab until this. So Mylar will continually check ComicVine for new issue information (only for those issues in the Upcoming section), and when it's available it will automatically populate it into the comic details section, mark it as Wanted and then remove it from the Upcoming section and it will appear in the Wanted section. Once in the Wanted section, it will perform searches for these Wanted issues every 6 hours by default (or if you have rss feeds enabled, it will also check the feeds every X minutes for new releases).

**Comment by AtariBaby** [comment]
Great!

**Comment by evilhero** [comment]
I figured out what I was trying to do with the getdefaultlocale - it involved some changes I was trying to incorporate that involved being able to switch the type of db that Mylar would use. It was never finished (it got too problematic), so I just removed it. 

Even though the OP got the issue fixed by adding the LC, the reference that was giving the error will be gone in the next development commit ;)
